main.c

```diff
@@ -1,52 +1,54 @@
-#include <xboxrt/debug.h>
+#include <hal/debug.h>
 #include <pbkit/pbkit.h>
 #include <hal/video.h>
 #include <hal/xbox.h>
-#include <hal/fileio.h>
+#include <winapi/fileapi.h>
+#include <winapi/handleapi.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <windows.h>
 #include "output.h"
 #include "global.h"
 #include "func_table.h"
 #include "vector.h"
 #include "string_extra.h"
 
 int load_conf_file(char *config_file_path) {
-    int handle;
-    unsigned int file_size = 0;
-    char *buffer;
-    unsigned int read = 0;
-    int result;
-
     print("Trying to open config file: %s", config_file_path);
-
-    result = XCreateFile(&handle,
+    HANDLE handle = CreateFile(
         config_file_path,
         GENERIC_READ,
         FILE_SHARE_READ,
+        0,
         OPEN_EXISTING,
-        FILE_ATTRIBUTE_NORMAL);
-    if(result != 0) {
+        FILE_ATTRIBUTE_NORMAL,
+        NULL
+    );
+    if(handle == INVALID_HANDLE_VALUE) {
         print("Could not open config file '%s' for read", config_file_path);
         return -1;
     }
 
-    XGetFileSize(handle, &file_size);
+    DWORD file_size = GetFileSize(handle, NULL);
+    if(file_size == INVALID_FILE_SIZE) {
+        print("ERROR: Could not get file size for %s", config_file_path);
```

Comment:
If this error is hit, the next malloc would likely fail. You should probably `return -1`